tortoise/backends/sqlite/executor.py

```diff
@@ -1,8 +1,33 @@
+from decimal import Decimal
+
 from pypika import Table
 
+from tortoise import fields
 from tortoise.backends.base.executor import BaseExecutor
 
 
+def to_db_bool(self, value):
+    if value is None:
+        return None
+    return int(bool(value))
+
+
+def to_db_decimal(self, value):
+    if value is None:
+        return None
+    if self.decimal_places == 0:
+        quant = '1'
+    else:
+        quant = '1.{}'.format('0' * self.decimal_places)
+    return Decimal(value).quantize(Decimal(quant)).normalize()
+
+
+TO_DB_OVERRIDE = {
+    fields.BooleanField: to_db_bool,
+    fields.DecimalField: to_db_decimal,
+}
```

Comment:
Such way to implement custom behaviour for fields in drivers seems quite hacky and not flexible enough, because it only allows to override already existing methods.
How about we do it this way:

- In module of driver you implement custom fields inheriting from their basic parent (Like `PostgresJSONField(fields.JSONField):`
- You create a mapping of `{<Base class>: <Driver custom class>}` somewhere in module
- On `Tortoise.init()` you patch models based on their driver class  replacing original class with a class from driver app

There is one more tricky moment with this strategy - different drivers can have different name for field type in db. To have better control over it - fields themselfs should have method like `get_db_type()` which will return their type definition. This way we can remove `FIELD_TYPE_MAP` from `SchemaGenerator` and users will be able to define custom fields with custom db type. This way we can also remove hardcode like this from `SchemaGenerator`:

```
if isinstance(field_object, fields.DecimalField):
    field_type = field_type.format(field_object.max_digits, field_object.decimal_places)
elif isinstance(field_object, fields.CharField):
    field_type = field_type.format(field_object.max_length)
```

and replace it with graceful:

```
    def get_db_type(self):
        return 'DECIMAL({},{})'.format(self.max_digits, self.decimal_places)
```

Reply:
Yeah, I was thinking it would be better to have the DB driver replace the Field instances with their own versions on create. Then all the custom logic could be contained.
The issue is there is a LOT of refactoring that would be required for that. So I wasn't sure what to do with it.

If we want to clean up the code structure, there is a lot of annoyances that are going to get in the way:
1) The logic is currently spread out a bit already
2) There are quite a few circular imports
3) There is a fair amount of duplication.
I should probably do some more investigation around this.